R/string_boundary_linter.R

```diff
@@ -98,12 +98,19 @@ string_boundary_linter <- function(allow_grepl = FALSE) {
 
   get_regex_lint_data <- function(xml, xpath) {
     expr <- xml_find_all(xml, xpath)
+    if (length(expr) == 0L) {
```

Comment:
Why short-circuit? This adds extra complexity later in the code.

Reply:
I guess the efficiency benefit is not much. Reverted